core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -2531,19 +2533,45 @@ wasi_ssp_sock_connect(wasm_exec_env_t exec_env, struct fd_table *curfds,
     struct fd_object *fo;
     __wasi_errno_t error;
     int ret;
+    printf("[SSP] sock connect\n");
+    
+    uint8_t *adjusted_addr = (uint8_t *)addr + 4;
+    printf("[SSP] wasi addr moved: %u.%u.%u.%u\n", adjusted_addr[0],
+                                                adjusted_addr[1],
+                                                adjusted_addr[2],
+                                                adjusted_addr[3]);
+    // use a copy because addr will be modified
+    uint8_t adjusted_addr_copy[4];
+    memcpy(adjusted_addr_copy, adjusted_addr, 4);
+
+    uint16_t *adjusted_port = (uint16_t *)(adjusted_addr + 4);
+    printf("[SSP] raw addr port %u\n", ntohs(*adjusted_port));
+
+    uint16_t adjusted_port_copy = ntohs(*adjusted_port);
+    
+    // adjust addr
+    addr->addr.ip4.addr.n0 = adjusted_addr_copy[3];
+    addr->addr.ip4.addr.n1 = adjusted_addr_copy[2]; 
+    addr->addr.ip4.addr.n2 = adjusted_addr_copy[1];
+    addr->addr.ip4.addr.n3 = adjusted_addr_copy[0];
+    addr->addr.ip4.port = adjusted_port_copy;
```

Comment:
Not sure why need to adjust the ip addr and port, is it possible to convert them before calling these functions? If needed, had better wrap a function and call it instead, and also control the code with zephyr macro?